protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java

```diff
@@ -288,6 +288,7 @@ public void resetIndex(long index, long timestamp) {
     currentOperation = OperationType.COMMAND;
     operationIndex = index;
     currentIndex = index;
+    commitIndex = index;
```

Comment:
not super sure about this.

Reply:
I think @johnou is right. The `commitIndex` should never be decreased, so this should be ensuring that doesn't happen with `setCommitIndex` rather than a simple update.